### PR TITLE
Interactivity API: Add missing directives and move e2e tests

### PR DESCRIPTION
**Pull request by DAreRodz**
## What?
Move the whole Interactivity API from [block-interactivity-experiments](https://github.com/WordPress/block-interactivity-experiments/), except those changes related to the client-side navigation feature.

Tracking issue: https://github.com/WordPress/gutenberg/issues/51056

## Why?
To complete the Interactivity API publicly testable version.

## How?
- [x] Add missing directives
- [x] Move the e2e tests from the old repo.

## Next

I decided to leave the following tasks for subsequent PRs:

- Add missing e2e tests.
- Add missing JSDocs to the code.

### CODE REVIEW

**Comment by luisherranz** [comment]
Hey David, it'd be great to switch to the `--` syntax in this branch as well.

I already made the SSR work with that syntax: https://github.com/WordPress/gutenberg/pull/51229

When do you want to do it? Should we do a new PR to modify the current code and then you modify this one? Or do you want to it in this PR? Or do you want to merge this PR first and then do another PR to modify everything?

**Comment by luisherranz** [comment]
If you don't mind, I think I'm going to go ahead and do a PR to do the syntax change.

Also, if you don't mind, I want to include `wp-text` in that PR because I want to start using it, but I'll let you add the e2e tests here.

Finally, I didn't add `wp-html` to the [SSR PR](https://github.com/WordPress/gutenberg/pull/51229) because it has security implications and the final name is not decided yet. We can leave it for later.

File: packages/e2e-tests/plugins/interactive-blocks/directive-bind/view.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+		actions: {
+			toggle: ( { state, foo } ) => {
+				state.url = '/some-other-url';
+				state.checked = ! state.checked;
+				state.show = ! state.show;
+				state.width += foo.bar;
+			},
+		},
+	} );
+} )( window );
```
Comment (luisherranz) [inline packages/e2e-tests/plugins/interactive-blocks/directive-bind/view.js:23]: Why do you need to keep the scope of this hidden in a function?

File: packages/e2e-tests/plugins/interactive-blocks/directive-bind/block.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+	"name": "test/directive-bind",
+	"title": "E2E Interactivity tests - directive bind",
+	"category": "text",
+	"icon": "heart",
+	"description": "",
+	"supports": {
+		"interactivity": true
+	},
+	"textdomain": "e2e-interactivity",
+	"viewScript": ["directive-bind-view", "wp-interactivity"],
```
Comment (luisherranz) [inline packages/e2e-tests/plugins/interactive-blocks/directive-bind/block.json]: Why do you need to add `wp-interactivity` here if you are adding it as a dependency when you register the script?

**Review by luisherranz — APPROVED** [review]
Looks good to me, David 🙂 

I just have two small questions, added below.

File: packages/e2e-tests/plugins/interactive-blocks/directive-bind/view.js (hunk 1 above)
Comment (DAreRodz) [reply packages/e2e-tests/plugins/interactive-blocks/directive-bind/view.js:23]: I replicated the same format other scripts have in the `plugins` folder. I guess it's not needed in our case, as the only thing we do is to call functions (`store()` or `directive()`).

The only exception is [this file](https://github.com/WordPress/gutenberg/pull/51194/files#diff-a8a1ec42aeab6e537c359b50e765a188cd1222cae4e5c7d1d13447d347cdee53).

File: packages/e2e-tests/plugins/interactive-blocks/directive-bind/block.json (hunk 2 above)
Comment (DAreRodz) [reply packages/e2e-tests/plugins/interactive-blocks/directive-bind/block.json]: IIRC, I had problems with script registering and enqueuing and tested different combinations. 😅 Let me check if that's required or not (it shouldn't, as you mentioned).

Comment (DAreRodz) [reply packages/e2e-tests/plugins/interactive-blocks/directive-bind/block.json]: Correct, it's not needed. I removed `wp-interactivity` from all `viewScript` props.

Comment (luisherranz) [reply packages/e2e-tests/plugins/interactive-blocks/directive-bind/block.json]: Ok, thank you!

File: packages/e2e-tests/plugins/interactive-blocks/directive-bind/view.js (hunk 1 above)
Comment (luisherranz) [reply packages/e2e-tests/plugins/interactive-blocks/directive-bind/view.js:23]: Ok, no problem 🙂

**Pull request merged**